clients/lsp-javascript.el

```diff
@@ -118,6 +118,12 @@ exceed allowed memory usage."
   :group 'lsp-typescript
   :type 'string)
 
+(defcustom lsp-clients-typescript-prefer-use-project-ts-server nil
```

Comment:
What is the vscode default here? If it is true I prefer going with t as default.

Reply:
> VS Code ships with a recent stable version of the TypeScript language service and uses this by default to provide IntelliSense in your workspace. 
>
> - [Using newer TypeScript versions](https://code.visualstudio.com/docs/typescript/typescript-compiling#_using-newer-typescript-versions)

Based on that, I believe that the VS Code default corresponds to our `nil`. 

This makes sense to me from the "just get something that works" perspective. VS Code (and lsp-mode) can install a newer and ideally known-working version of typescript (and the language server) and use those. Users who need something more special can step outside that.

Reply:
Sounds good. Feel free to convert it from draft to PR.